packages/notifi-wallet-provider/lib/hooks/useBinance.ts

```diff
@@ -0,0 +1,192 @@
+    const timer = setTimeout(() => {
+      disconnectWallet();
+      loadingHandler(false);
+    }, timeoutInMiniSec ?? 5000);
+
+    try {
+      const response: { addresses: { type: string; address: string }[] }[] =
+        await provider.requestAccounts();
+      const address = response?.[0]?.addresses?.find((v) => v.type === 'eth')
+        ?.address as string;
+
+      const walletKeys = {
+        bech32: converter('inj').toBech32(address), // TODO: dynamic cosmos chain addr conversion
+        hex: address,
+      };
+
+      selectWallet(walletName);
+      setWalletKeys(walletKeys);
+      setWalletKeysToLocalStorage(walletName as keyof Wallets, walletKeys);
+      return walletKeys;
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    } catch (e: any) {
+      console.error(e);
+      disconnectWallet();
+      e.message && errorHandler(new Error(e.message));
+      return null;
+    } finally {
+      loadingHandler(false);
+      clearTimeout(timer);
+    }
+  };
+
+  const disconnectWallet = () => {
+    setWalletKeys(null);
+    cleanWalletsInLocalStorage();
+    selectWallet(null);
+  };
+
+  const signArbitrary = useCallback(
+    async (message: string): Promise<`0x${string}` | undefined> => {
+      if (!provider || !walletKeys) {
+        handleWalletNotExists('signArbitrary');
+        return;
+      }
+
+      loadingHandler(true);
+      const timer = setTimeout(() => {
+        loadingHandler(false);
+      }, 5000);
+
+      try {
+        const signature: Promise<`0x${string}`> = await provider.request?.({
+          method: 'personal_sign',
+          params: [
+            Buffer.from(message).toString('hex'),
+            /** ⬆️
+             * hex-encoded UTF-8 string to present to the user. See how to encode a string like this in the browser-string-hexer module.
+             * @ref https://docs.metamask.io/wallet/reference/personal_sign/
+             */
+            walletKeys?.hex,
+          ],
+        });
+
+        return signature;
+        // ⬆️ Note:A hex-encoded 129-byte array starting with 0x.
+      } catch (e) {
+        errorHandler(
+          new Error(
+            `${walletName}'s signArbitrary failed, check console for details`,
+          ),
+        );
+        console.error(e);
+      } finally {
+        loadingHandler(false);
+        clearTimeout(timer);
+      }
+    },
+    [walletKeys?.hex],
+  );
+
+  return {
+    walletKeys,
+    isWalletInstalled,
+    connectWallet,
+    signArbitrary,
+    disconnectWallet,
+  };
+};
+
+const geBinanceFromWindow = async (): Promise<BinanceChain> => {
+  if (typeof window === 'undefined' || !window.BinanceChain) {
+    throw new Error(
+      `Cannot get ${walletName} without a window | Cannot get ${walletName} from window`,
+    );
+  }
+
+  if (window.BinanceChain) {
+    return window.BinanceChain;
+  } else if (document.readyState === 'complete') {
+    throw new Error(`Please install the ${walletName} extension`);
```

Comment:
We'll want to remove this error message per ODF.

Reply:
yeah, I've taken care of that. The error message won't be visible on the frontend unless we specifically need it. 
this way, it remains available for troubleshooting.